internal/services/elastic/elastic_tag_rule_resource.go

```diff
@@ -0,0 +1,308 @@
+		Read:   resourceElasticTagRuleRead,
+		Update: resourceElasticTagRuleCreateorUpdate,
+		Delete: resourceElasticTagRuleDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ElasticTagRuleID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ElasticMonitorName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"rule_set_name": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  utils.String("default"),
+			},
+
+			"id": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"type": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"log_rules": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"send_aad_logs": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+
+						"send_subscription_logs": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+
+						"send_activity_logs": {
+							Type:     pluginsdk.TypeBool,
+							Optional: true,
+						},
+
+						"filtering_tag": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"name": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+									},
+									"value": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+									},
+									"action": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											"Include",
+											"Exclude",
+										}, false),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"provisioning_state": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceElasticTagRuleCreateorUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).Elastic.TagRuleClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	ruleSetName := d.Get("rule_set_name").(string)
+
+	id := parse.NewElasticTagRuleID(subscriptionId, resourceGroup, name, ruleSetName).ID()
+
+	existing, err := client.Get(ctx, resourceGroup, name, ruleSetName)
```

Comment:
This function is used in both `create` and `update`, so we need to know whether this is a new resource by `d.IsNewResource()`, because if it's a new resource, it must not exist otherwise we'll throw `ImportAsExistsError`.

```
       if d.IsNewResource() {
		existing, err := client.Get(ctx, id.ResourceGroup, id.MonitorName, id.RuleSetName)
		if err != nil {
			if !utils.ResponseWasNotFound(existing.Response) {
				return fmt.Errorf("checking for existing %q: %+v", id, err)
			}
		}
		if !utils.ResponseWasNotFound(existing.Response) {
			return tf.ImportAsExistsError("azurerm_elastic_tag_rule", id.ID())
		}
	}
```